### ISSUE TITLE
Consider adding warnings when a DL element contains duplicate DT names

### ISSUE THREAD

**Issue by EricDunsworth** [validator/validator#1662]
**URL of WHATWG spec or PR:**
https://html.spec.whatwg.org/multipage/grouping-content.html#the-dl-element

**Excerpt:**
> Within a single [dl](https://html.spec.whatwg.org/multipage/grouping-content.html#the-dl-element) element, there should not be more than one [dt](https://html.spec.whatwg.org/multipage/grouping-content.html#the-dt-element) element for each name.

**Describe the solution you'd like:**
I wanted to propose showing validation warnings (similar to duplicate ID errors) when a ``dl`` element contains multiple ``dt`` elements with identical names.

**Examples of potential warning messages:**
> **Warning:** Duplicate ``dt`` name in ``dl`` element. Consider using unique ``dt`` names. Within a single [dl](https://html.spec.whatwg.org/multipage/grouping-content.html#the-dl-element) element, there should not be more than one [dt](https://html.spec.whatwg.org/multipage/grouping-content.html#the-dt-element) element for each name.

> **Warning:** The first occurrence of ``dt`` ``[DT's name]`` was here.

**Affected code sample ( ``dt`` elements contain redundant names):**
```html
<!DOCTYPE html>
<html lang="">
<head>
<title>Test</title>
</head>
<body>

<dl>
	<dt>Term</dt>
	<dd>Description</dd>

	<dt>Term</dt>
	<dd>Description</dd>
</dl>

</body>
</html>
```

**Unaffected code sample (``dt`` elements contain unique names):**
```html
<!DOCTYPE html>
<html lang="">
<head>
<title>Test</title>
</head>
<body>

<dl>
	<dt>Term 1</dt>
	<dd>Description</dd>

	<dt>Term 2</dt>
	<dd>Description</dd>
</dl>

</body>
</html>
```

**PS:**
Not sure where a parent ``<dl>`` containing a child ``dl`` with a redundant ``dt`` name would stand...

**Uncertain code sample (parent and child ``dl``s contain the same ``dt`` name):**
```html
<!DOCTYPE html>
<html lang="">
<head>
<title>Test</title>
</head>
<body>

<dl>
	<dt>Term</dt>
	<dd>
		<dl>
			<dt>Term</dt>
			<dd>Description</dd>
		</dl>
	</dd>
</dl>

</body>
</html>
```

**Comment by sideshowbarker** [comment]
Implemented and deployed:

https://validator.w3.org/nu/?showsource=yes&doc=data%3Atext%2Fhtml%3Bcharset%3Dutf-8%3Bbase64%2CPCFET0NUWVBFIGh0bWw%2BDQo8aHRtbCBsYW5nPSIiPg0KPGhlYWQ%2BDQo8dGl0bGU%2BVGVzdDwvdGl0bGU%2BDQo8L2hlYWQ%2BDQo8Ym9keT4NCg0KPGRsPg0KCTxkdD5UZXJtPC9kdD4NCgk8ZGQ%2BRGVzY3JpcHRpb248L2RkPg0KDQoJPGR0PlRlcm08L2R0Pg0KCTxkZD5EZXNjcmlwdGlvbjwvZGQ%2BDQo8L2RsPg0KDQo8L2JvZHk%2BDQo8L2h0bWw%2B
